custom/enikshay/reports/views.py

```diff
@@ -73,6 +75,23 @@ def duplicate_ids_report(request, domain, case_type):
     counts = Counter(case['readable_id'] for case in all_cases)
     bad_cases = [case for case in all_cases if counts[case['readable_id']] > 1]
 
+
+    for case in bad_cases:
+        form = CaseAccessorSQL.get_transactions(case['case_id'])[0].form
+        if form:
+            case['form_name'] = form.form_data.get('@name', 'NA')
+            form_device_number = form.form_data.get('serial_id', {}).get('outputs', {}).get('device_number')
+            case['device_number_in_form'] = form_device_number
+
+            user = CommCareUser.get(form.user_id) if form.user_id else None
```

Comment:
is this a temporary report or something that needs to scale? these one-off lookups will be expensive at scale and would benefit from in-memory caching

Reply:
This is a temporary report used only internally (for identifying issues with the eNikshay ID generation scheme).  It is fairly slow, but that's fine as long as it doesn't negatively impact other HQ functionality.